modules/lsfilter/models/lsfilter_saved_queries.php

```diff
@@ -127,20 +127,26 @@ public static function save_query( $name, $query, $scope ) {
 				throw new Exception("Unknown scope");
 		}
 
-		/* It should be an update on duplicate key, but that doesn't work well with null-valued columns, delete instead */
+		// Check if exist
 		if( $user === null ) {
-			$sql_query = "DELETE FROM ".self::tablename." WHERE username IS NULL AND filter_name = %s";
-			$args = array($name);
+			$sql_query = "SELECT `id` FROM ".self::tablename." WHERE username IS NULL AND filter_name = %s LIMIT 1";
```

Comment:
Why do we need LIMIT here? Does this mean we exppect multiple entriies for each filter?

Reply:
filter_name is unique and does not expect multiple entry. Just want to make sure that a single id is returned in case there's a multiple entry outside gui process for testing.